app/editor/components/MentionMenu.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { v4 } from "uuid";
+import { MenuItem } from "@shared/editor/types";
+import useRequest from "~/hooks/useRequest";
+import useStores from "~/hooks/useStores";
+import CommandMenu, { Props } from "./CommandMenu";
+import MentionMenuItem from "./MentionMenuItem";
+
+interface MentionItem extends MenuItem {
+  name: string;
+  title: string;
+  attrs: {
+    "data-type": string;
+    "data-id": string;
+    label: string;
+  };
+}
+
+type MentionMenuProps = Omit<
+  Props<MentionItem>,
+  "renderMenuItem" | "items" | "onLinkToolbarOpen" | "embeds" | "onClearSearch"
+>;
+
+function MentionMenu({ search, ...rest }: MentionMenuProps) {
+  const [items, setItems] = React.useState<MentionItem[]>([]);
+  const { users, auth } = useStores();
+  const { data, request } = useRequest(
+    React.useCallback(() => users.fetchPage({ query: search }), [users, search])
+  );
+
+  React.useEffect(() => {
+    request();
+  }, [request]);
+
+  React.useEffect(() => {
+    if (data) {
+      setItems(
+        data.map((d) => ({
+          name: "mention",
+          title: d.name,
+          appendSpace: true,
+          attrs: {
+            "data-type": "user",
+            "data-id": d.id,
+            "data-actor": auth.user?.id,
```

Comment:
These don't need "data-" in front, they represent the attributes stored in the document data and don't have to directly map to dom properties.

```suggestion
            type: MentionType.User,
            id: d.id,
            actorId: auth.user?.id,
```